### PR TITLE
SOLR-15171 Add redirects from old Solr site to new

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-15171

These are the Lucene-side redirects to new Solr site

### CODE REVIEW

File: themes/lucene/templates/htaccess.template [hunk 1]
```diff
@@ -126,19 +91,24 @@ RedirectMatch Permanent ^/nutch/(.*)$ https://nutch.apache.org/$1
...
+# Redirect old location of Solr release doc and javadoc to new, changed location in solr.apache.org/docs
+RedirectMatch temp ^/solr/api$ https://solr.apache.org/api/
+RedirectMatch Permanent ^/solr/(\d+_\d+_\d+)$ https://solr.apache.org/docs/$1
+# These will be further rewritten to SOLR_LATEST_RELEASE on the solr side
+RedirectMatch temp ^/solr/api/org/(.*) https://solr.apache.org/api/org/$1
+RedirectMatch temp ^/solr/api/(.*) https://solr.apache.org/api/$1
+RedirectMatch Permanent ^/solr/4_0_0-(ALPHA|BETA)(.*) https://solr.apache.org/docs/4_0_0/$2
+RedirectMatch Permanent ^/solr/api-(.*) https://solr.apache.org/docs/$1
+# Fallback redirect for anythin not caught above
+RedirectMatch Permanent ^/solr/(.*)$ https://solr.apache.org/$1
```
Comment (janhoy) [inline themes/lucene/templates/htaccess.template:104]: I deleted all solr related redirects from this file and added the needed ones in one single block here.
The strategy is to first catch specialized patterns that need custom handling, and then fallback to the generic `/solr/.*` wildcard redirect, which may then trigger further rules on the Solr-side.

* First we tackle some javadoc locations and send to the new solr.apache.org/docs/ location
* The ref guide URLs `/solr/guide/.*` as well as `/solr/doap.rdf` will be covered by the last fallback rule

Comment (uschindler) [reply themes/lucene/templates/htaccess.template:106]: This is all much too complicated! Why do we need the specializations at all. In addition they still contain duplicates (caused by the "__root"-aapend-slash-handling added on the solr side.

We only need exactly one redirect; it should just replace prefix:

```
Redirect permanent /solr https://solr.apache.org
``` 

Everything else is handled by Solr's side. Browsers are perfectly happy with 2 or three redircets. Also the main purpose here is to tell lucene.apache.org/solr visitors that the main "permanent" target is solr.apache.org. Later redirects should be done from there, as it's on Solr's responsibility.

We we also need no wildcard redirect (regex not needed). The above plain standard prefix redirect is enough. Also order of "standard prefix" redirects in apache is by length. Longest prefixes first.

https://httpd.apache.org/docs/2.4/mod/mod_alias.html#redirect: "Then any request beginning with URL-path will return a redirect request to the client at the location of the target URL. Additional path information beyond the matched URL-path will be appended to the target URL."

You can test all this on the Lucene staging site.

**Review by uschindler — CHANGES_REQUESTED** [review]

File: themes/lucene/templates/htaccess.template (hunk 1 above)
Comment (janhoy) [reply themes/lucene/templates/htaccess.template:106]: Ok, nice to know. Will make another attempt...

File: themes/lucene/templates/htaccess.template [hunk 2]
```diff
@@ -126,19 +91,15 @@ RedirectMatch Permanent ^/nutch/(.*)$ https://nutch.apache.org/$1
 # Lucy has moved, site strcuture has changed, do don't use $1 in redirect...
 RedirectMatch Permanent ^/lucy/(.*)$ https://lucy.apache.org/
 
-# promote Solr to TLP
-# TODO: This links to the staging site for new Solr - change the redirect to TLP page
-#RedirectMatch Permanent ^/solr/(.*)$ https://solr.apache.org/$1
-RedirectMatch Permanent ^/solr/(.*)$ https://lucene-solrtlp.staged.apache.org/$1
-
+# Solr moved to TLP Feb 2021
```
Comment (janhoy) [inline themes/lucene/templates/htaccess.template:94]: Simplified as two rules - needs separate rule for the ATOM feed

Comment (uschindler) [reply themes/lucene/templates/htaccess.template:94]: This also simplifies the maintenance. Lucene webserver maintainer should not need to maintain those rules.

Comment (uschindler) [reply themes/lucene/templates/htaccess.template:94]: Did not know about that feeds URL. +1

**Review by uschindler — APPROVED** [review]

**Comment by uschindler** [comment]
Hi, I think we still need one redirect on the old page using match:

`RedirectMatch permanent ^/solr/(\d+_\d+_\d+\b.*)$ https://solr.apache.org/docs/$1`

Alternatively we should add a redirect on Solr's side. This is indeed a special case, because there's nothing on Solr's side that handles this.

This redirect should be placed before the generic one.

**Comment by uschindler** [comment]
Refguide is fine, as this does not need the version number patterns. Also the "/api" ones are fine.

**Comment by janhoy** [comment]
> Alternatively we should add a redirect on Solr's side. This is indeed a special case, because there's nothing on Solr's side that handles this.

I just committed what I believe was needed, before your comment here: https://github.com/apache/lucene-site/commit/35fedb17f244b44e6197defc2e92c4fe2e9c9b4b

**Pull request merged**